src/Microsoft.AspNetCore.Razor.Language/RazorSourceDocument.cs

```diff
@@ -201,7 +201,37 @@ public static RazorSourceDocument ReadFrom(RazorProjectItem projectItem)
         /// <returns>The <see cref="RazorSourceDocument"/>.</returns>
         /// <remarks>Uses <see cref="System.Text.Encoding.UTF8" /></remarks>
         public static RazorSourceDocument Create(string content, string fileName)
-            => Create(content, fileName, Encoding.UTF8);
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
+            return Create(content, fileName, Encoding.UTF8);
```

Comment:
`=>` is evil if you have to have a newline. In this case someone went for the pranav points and forgot about argument validation.